packages/babel-parser/src/parser/util.js

```diff
@@ -158,7 +161,10 @@ export default class UtilParser extends Tokenizer {
       messageOrType = `Unexpected token, expected "${messageOrType.label}"`;
     }
     /* eslint-disable @babel/development-internal/dry-error-messages */
-    throw this.raise(pos != null ? pos : this.state.start, messageOrType);
+    throw this.raise(pos != null ? pos : this.state.start, {
+      code: "UnexpectedToken",
+      template: messageOrType,
+    });
```

Comment:
I think here it would be better if the signature was
```js
unexpected(pos: ?number, messageOrType: TokenType | ErrorTemplate)
```

and if we get an error template we use the correct code rather than `UnexpectedToken`.